src/components/CustomRegisterField/CustomRegisterField.tsx

```diff
@@ -0,0 +1,70 @@
+import { type FC, type ChangeEventHandler, type ReactNode, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import type { GetRegisterFieldOption } from '@inplayer-org/inplayer.js';
+
+import type { CustomRegisterFieldVariant } from '#types/account';
+import { isTruthyCustomParamValue, testId } from '#src/utils/common';
+import Checkbox from '#components/Checkbox/Checkbox';
+import TextField from '#components/TextField/TextField';
+import Radio from '#components/Radio/Radio';
+import Dropdown from '#components/Dropdown/Dropdown';
+import DateField from '#components/DateField/DateField';
+
+type Props = {
+  type?: CustomRegisterFieldVariant;
+  name: string;
+  value: string | boolean;
+  onChange: ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
+} & Partial<{
+  label: ReactNode;
+  placeholder: string;
+  error: boolean;
+  helperText: string;
+  disabled: boolean;
+  required: boolean;
+  options: GetRegisterFieldOption;
+}>;
+
+export type CustomRegisterFieldCommonProps = Props;
+
+export const CustomRegisterField: FC<Props> = ({ type, value = '', ...props }) => {
+  const { t, i18n } = useTranslation();
+
+  const optionsList = useMemo(() => {
+    switch (type) {
+      case 'country':
+      case 'us_state': {
+        const codes = Object.keys(i18n.getResourceBundle(i18n.language, type));
+
+        return codes.map((code) => ({
+          value: code,
+          label: t(`${type}:${code}`),
+        }));
+      }
+      default: {
+        if (props.options) {
+          return Object.entries(props.options).map(([value, label]) => ({ value, label }));
+        }
+
+        return [];
+      }
+    }
+  }, [t, type, props.options, i18n]);
```

Comment:
This is a matter of personal preference.

- Sometimes I like to create a separate var, such as `const codes`, so that its name would be descriptive enough and would be make it more apparent to other developers (and to my future self) what I'm doing - in this case I'm trying to make it more obvious why I need `Object.keys(i18n.getResourceBundle(i18n.language, type))` - it's for the (country or US state) codes. So I'd rather have us keep that const.

- On the `if` vs `switch` state - I'm in favor of either `switch` or the `[].includes` pattern.

- On `Object.entries(options || []).map()` vs `if (!options) return [];` - the latter would prevent `.map()` to be run at all if `!options`. A teeny tiny optimisation.